libraries/core_libs/consensus/src/final_chain/final_chain.cpp

```diff
@@ -308,8 +310,10 @@ class FinalChainImpl final : public FinalChain {
       chunk_to_alter[index % c_bloomIndexSize] |= log_bloom_for_index;
       db_->insert(batch, DB::Columns::final_chain_log_blooms_index, chunk_id, util::rlp_enc(rlp_strm, chunk_to_alter));
     }
-    db_->insert(batch, DB::Columns::final_chain_transaction_hashes_by_blk_number, blk_header.number,
-                dev::rlp(hashes_from_transactions(transactions)));
+    if (!kLightNode) {
+      db_->insert(batch, DB::Columns::final_chain_transaction_hashes_by_blk_number, blk_header.number,
```

Comment:
Maybe better way to save time and storage on this is to totally remove this column and get this from PeriodData. As it is used for rpc and in some cases it is overcomplicated. So its getting hashes from this column and then getting transactions one by one for the whole block. But we can just get all transactions from PeriodData